libgalois/src/analytics/connected_components/connected_components.cpp

```diff
@@ -688,16 +688,33 @@ struct ConnectedComponentsAfforestAlgo {
       : plan_(plan) {}
 
   void Initialize(Graph* graph) {
+    //katana::do_all(katana::iterate(*graph), [&](const GNode& node) {
+    //  graph->GetData<NodeComponent>(node) = new NodeAfforest();
+    //});
+    // XXX
+    //int sum = 0;
     katana::do_all(katana::iterate(*graph), [&](const GNode& node) {
-      graph->GetData<NodeComponent>(node) = new NodeAfforest();
+      auto& snode = graph->GetData<NodeComponent>(node);
+      new (&snode) NodeAfforest();
```

Comment:
Was the type of `snode` changed? I don't understand the change is equivalent.

Reply:
I don't think I quite understand your question

Reply:
I didn't know about in-place new. So this is not allocating new memory but just calling constructor on it (initializing it)?

Reply:
Is this still needed? 

Please drop commented code as well.

Reply:
Yes, what you said about in-place new is correct. It calls a constructor on the pointer you provide.

I wanna keep the old code until we fix the Numa problems with our node data. Then, if we're happy, we can easily go back.